### PR TITLE
Clean up function bindings

### PR DESCRIPTION
**Pull request by jl-wynen**
Fixes a number of issues with the bindings and adds more tests. I think we have coverage of all exposed functions now.

Also removes some unused bindings.

### CODE REVIEW

File: docs/reference/free-functions.rst [hunk 1]
```diff
@@ -59,6 +60,7 @@ Math
    round
    sqrt
    subtract
+   unary_minus
```
Comment (SimonHeybrock) [inline docs/reference/free-functions.rst:63]: Numpy calls this differently? https://numpy.org/doc/stable/reference/generated/numpy.negative.html

File: src/scipp/core/logical.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    Equivalent to
+
+    .. code-block:: python
+
+        ~a
+
+    :param x: Input data.
+    :return: The logical inverse of `x`.
+    """
```
Comment (SimonHeybrock) [inline src/scipp/core/logical.py:22]: Remember that we can use a new docstring style! But we may also want to leave this here and try to auto-convert all of them at once.

Comment (jl-wynen) [reply src/scipp/core/logical.py:22]: Ah right, this was largely copy-pasted. I am going to change it, there is no reason to merge it like this.

File: docs/reference/free-functions.rst (hunk 1 above)
Comment (jl-wynen) [reply docs/reference/free-functions.rst:63]: That's why I didn't find it! Renaming...

File: src/scipp/core/logical.py (hunk 2 above)
Comment (jl-wynen) [reply src/scipp/core/logical.py]: Done. I tried pyment but it struggles with type hints like `Union`, example sections, and see also sections. I ended up doing it with a few search and replace operations...

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**